boot/bootutil/src/image_validate.c

```diff
@@ -497,7 +499,8 @@ bootutil_img_validate(struct enc_key_data *enc_state, int image_index,
              */
             fih_rc = fih_ret_encode_zero_equality(img_security_cnt <
                                    fih_int_decode(security_cnt));
-            if (fih_not_eq(fih_rc, FIH_SUCCESS)) {
+            if (FIH_NOT_EQ(fih_rc, FIH_SUCCESS)) {
+                FIH_SET(fih_rc, FIH_FAILURE);
```

Comment:
should not be line 529 the same? FIH_SET(fih_rc, FIH_FAILURE);

What about the declaration of the fih_ret variables? Should not all `fih_rc` be initialised by FIH_SET?

Might a macro would handy:
`FIH_DECLEAR(x) (fih_ret x; FIH_SET(x, FIH_FAILURE));`